### PR TITLE
125 tws tests srv@main

### PR DESCRIPTION
**Pull request by BLAZEWIM**
Fixes #[125](https://github.com/insightsengineering/teal.widgets/issues/125)

Server tests for table with settings, similiar to https://github.com/insightsengineering/teal.widgets/issues/113

### CODE REVIEW

File: tests/testthat/test-table_with_settings.R [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      shiny::testServer(
+        teal.widgets:::table_with_settings_srv,
+        args = arg
+      ),
+      "Assertion"
+    )
+  }
+})
+
+testthat::test_that("table_with_settings_srv: hide works", {
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:24]: We are testing that it does not render any error but not if it works, correct?

File: tests/testthat/test-table_with_settings.R [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    )
+})
+
+testthat::test_that("table_with_settings_srv: produce plot html", {
+  shiny::testServer(
+    teal.widgets:::table_with_settings_srv,
+    args = list(id = "tws", table_r = table_r),
+    expr = {
+      expect_type(output$table_out_modal$html, "character")
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:39]:
```suggestion
      testthat::expect_s3_class(output$table_out_modal$html, "html")
```

File: tests/testthat/test-table_with_settings.R [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    shiny::testServer(
+      teal.widgets:::table_with_settings_srv,
+      args = list(table_r = table_r, show_hide_signal = reactive(FALSE)),
+      expr = {
+        expect_silent(output$table_out_modal$html)
+        }
+    )
+})
+
+testthat::test_that("table_with_settings_srv: produce plot html", {
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:34]:
```suggestion
testthat::test_that("table_with_settings_srv: return html table", {
```

File: tests/testthat/test-table_with_settings.R [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  )
+})
+
+testthat::test_that("table_with_settings_srv: expand works", {
+  shiny::testServer(
+    teal.widgets:::table_with_settings_srv,
+    args = list(id = "tws", table_r = table_r),
+    expr = {
+      session$setInputs(`expand` = TRUE)
+      expect_silent(output$table_out_modal$html)
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:51]:
```suggestion
      expect_silent(output$table_out_modal)
```

File: tests/testthat/test-table_with_settings.R [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    )
+  }
+})
+
+testthat::test_that("table_with_settings_srv: hide works", {
+    shiny::testServer(
+      teal.widgets:::table_with_settings_srv,
+      args = list(table_r = table_r, show_hide_signal = reactive(FALSE)),
+      expr = {
+        expect_silent(output$table_out_modal$html)
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:29]:
```suggestion
        testthat::expect_silent(output$table_out_modal$html)
```

File: tests/testthat/test-table_with_settings.R [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  )
+})
+
+testthat::test_that("table_with_settings_srv: expand works", {
+  shiny::testServer(
+    teal.widgets:::table_with_settings_srv,
+    args = list(id = "tws", table_r = table_r),
+    expr = {
+      session$setInputs(`expand` = TRUE)
+      expect_silent(output$table_out_modal)
```
Comment (mhallal1) [inline tests/testthat/test-table_with_settings.R:51]:
```suggestion
      testthat::expect_silent(output$table_out_modal)
```

File: tests/testthat/test-table_with_settings.R (hunk 1 above)
Comment (BLAZEWIM) [reply tests/testthat/test-table_with_settings.R:24]: Yes, and I don't know how to test it differently with `testServer`

Comment (mhallal1) [reply tests/testthat/test-table_with_settings.R:24]: Ok whether the plot is shown or not is to be tested by `shinytest2`. The intention here is that setting `show_hide_signal` to FALSE does not break the functionality and still returns a table html which is hidden.

**Pull request merged**